auditors/apparmor/apparmor.go

```diff
@@ -92,6 +99,31 @@ func auditContainer(container *k8stypes.ContainerV1, resource k8stypes.Resource)
 	return nil
 }
 
+func auditPodAnnotations(resource k8stypes.Resource, containerNames []string) []*kubeaudit.AuditResult {
+	var auditResults []*kubeaudit.AuditResult
+	for annotationKey, annotationValue := range k8s.GetAnnotations(resource) {
+		if !strings.HasPrefix(annotationKey, ContainerAnnotationKeyPrefix) {
+			continue
+		}
+		containerName := strings.Split(annotationKey, "/")[1]
+		if !contains(containerNames, containerName) {
+			auditResults = append(auditResults, &kubeaudit.AuditResult{
+				Name:     AppArmorInvalidAnnotation,
+				Severity: kubeaudit.Warn,
```

Comment:
Could we have `kubeaudit.Error`? kubeuadit will remove that annonation right (`&fix.ByRemovingPodAnnotation`)?

Reply:
Good catch! We've struggled with the severity level in the past because we don't have a definition for what should go in each category. Based on the current behaviour I think it could be described as

Error: A security issue
Warn: Best practice and also when Error results are overridden (maybe we should make a separate severity level for overridden results???)
Info: Currently only used by the image auditor to give "success" output (arguably we should get rid of this to be consistent with the rest of kubeaudit where no output means everything is good)

I would say this check isn't directly a security issue nor a best practice 🤔. But I think you're right in that it makes sense to be an error level result rather than a warning!

As a side note, we do have another auditor which produces a warning result with a pending fix. The severity level doesn't matter for autofixing.

Reply:
Yes, you're right. This check doesn't seem to fit either category (Error / Warn) 

What if we use `Info` for overriden results? Or `Notice`? 

Oh good to know! So autofixing doesn't relate to the severity level. Thanks for explaining! 🙏🏽